runtime/parachains/src/hrmp.rs

```diff
@@ -1036,6 +1128,8 @@ impl<T: Config> Pallet<T> {
 			config.hrmp_sender_deposit.unique_saturated_into(),
 		)?;
 
+		// mutating storage -- shall not bail henceforth.
```

Comment:
I think this comment is misleading since mutating storage happens in reserve.

Reply:
yeah I see your point. I guess what I am trying to say is that **mutating storage one way or another**. `reserve` is fallible, what follows this comment is no longer fallible.

Reply:
made it clear now.